src/Datadog.Trace.ClrProfiler.Managed/CallTarget/Handlers/EndMethodHandler`1.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Reflection.Emit;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler.CallTarget.Handlers.Continuations;
+#pragma warning disable SA1649 // File name must match first type name
+
+namespace Datadog.Trace.ClrProfiler.CallTarget.Handlers
+{
+    internal static class EndMethodHandler<TIntegration, TTarget, TReturn>
+    {
+        private static readonly InvokeDelegate _invokeDelegate = null;
+        private static readonly ContinuationGenerator<TIntegration, TTarget, TReturn> _continuationGenerator = null;
+
+        static EndMethodHandler()
+        {
+            Type returnType = typeof(TReturn);
+            try
+            {
+                DynamicMethod dynMethod = IntegrationMapper.CreateEndMethodDelegate(typeof(TIntegration), typeof(TTarget), returnType);
+                if (dynMethod != null)
+                {
+                    _invokeDelegate = (InvokeDelegate)dynMethod.CreateDelegate(typeof(InvokeDelegate));
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new CallTargetInvokerException(ex);
+            }
+
+            Type resultType = typeof(void);
+
+            if (returnType.IsGenericType)
+            {
+                resultType = ContinuationsHelper.GetResultType(returnType);
```

Comment:
Small optimization: the `resultType` is only used when we determine the `TReturn` is a `Task<>` or `ValueTask<>`, so you could move this line to be run inside the conditional blocks

Reply:
you're right, we can save that call for non `Task<>` and `ValueTask<>`. I'll do the change.

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1007/commits/91b2b1c4a3158d30160fa3373de190682e13eb83